### PR TITLE
feat: enable attention dispatch for huanyuan video

### PR DESCRIPTION
**Pull request by DefTruth**
enable attention dispatch for huanyuan video, adapted from: 

https://github.com/huggingface/diffusers/blob/ac5a1e28fc9cc233863bcfb2abb9eef6807f156f/src/diffusers/models/transformers/transformer_hunyuanimage.py#L46C7-L46C32

## Example

```python
import torch
from diffusers import HunyuanVideoPipeline
from diffusers.utils import export_to_video

pipe: HunyuanVideoPipeline = HunyuanVideoPipeline.from_pretrained(
    "hunyuanvideo-community/HunyuanVideo",
    torch_dtype=torch.bfloat16,
)

pipe.enable_model_cpu_offload()
pipe.vae.enable_tiling()

def run_pipe():
    prompt = "A cat walks on the grass, realistic"
    output = pipe(
        prompt,
        height=320,
        width=512,
        num_frames=61,
        num_inference_steps=30,
        generator=torch.Generator("cpu").manual_seed(0),
    ).frames[0]
    return output

video = run_pipe()
export_to_video(video, "huanyuan_video.mp4", fps=15)
```

|w/o this PR| w/ this PR|
|:---:|:---:|
|huanyuan_video_ori|huanyuan_video|

@yiyixuxu pipelines

### CODE REVIEW

File: src/diffusers/models/transformers/transformer_hunyuan_video.py [hunk 1]
```diff
@@ -81,46 +85,61 @@ def __call__(
             if attn.add_q_proj is None and encoder_hidden_states is not None:
                 query = torch.cat(
                     [
-                        apply_rotary_emb(query[:, :, : -encoder_hidden_states.shape[1]], image_rotary_emb),
-                        query[:, :, -encoder_hidden_states.shape[1] :],
+                        apply_rotary_emb(
```
Comment (sayakpaul) [inline src/diffusers/models/transformers/transformer_hunyuan_video.py:88]: What's happening here?

Comment (DefTruth) [reply src/diffusers/models/transformers/transformer_hunyuan_video.py:88]: Since transpose is no longer needed, at this point seq_len is located in dim=1.

Comment (DefTruth) [reply src/diffusers/models/transformers/transformer_hunyuan_video.py:88]: This preprocess is same as: https://github.com/huggingface/diffusers/blob/ac5a1e28fc9cc233863bcfb2abb9eef6807f156f/src/diffusers/models/transformers/transformer_hunyuanimage.py#L46C7-L46C32

**Comment by sayakpaul** [comment]
Thanks for your contributions!
